src/Executables/Benchmark/Benchmark.cpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <benchmark/benchmark.h>
+#include <vector>
+
+#include "DataStructures/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Numerical/Spectral/LegendreGaussLobatto.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Element.hpp"
+#include "PointwiseFunctions/MathFunctions/PowX.hpp"
+// #include "Functors/Gradient.hpp"
+#include "Domain/GridCoordinates.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+// This file is an example of how to do microbenchmark with Google Benchmark
+// https://github.com/google/benchmark
+// For two examples in different anonymous namespaces
+
+namespace {
+// Benchmark of push_back() in std::vector, following Chandler Carruth's talk
+// at CppCon in 2015,
+// https://www.youtube.com/watch?v=nXaxk27zwlk&feature=youtu.be
+
+static void bench_create(benchmark::State &state) {
+ while (state.KeepRunning()) {
```

Comment:
According to the benchmark documentation, using `for(auto _ : state)` is preferred over using `state.KeepRunning()` which is available since benchmark version v1.2.0, and even though it's not what's used in Chandler's talk our example should probably use the most up to date form. This isn't the most up to date version available in spack though, which is v1.1.0.

Reply:
Spack has 1.2

Reply:
This doesn't work.

Reply:
See the documentation for 1.2: https://github.com/google/benchmark/tree/v1.2.0

Reply:
Ok I got the versions wrong, since 1.2 is the most up to date tag they have which is of course behind master. I'm fine conforming with 1.2 then since the change from KeepRunning -> auto _ was only committed 10 days ago.